packages/react/src/utils/query.ts

```diff
@@ -0,0 +1,140 @@
+import { hashFn } from '@bigmi/client'
+import type { Compute, ExactPartial, UnionStrictOmit } from '@bigmi/core'
+import {
+  type DefaultError,
+  type QueryKey,
+  type UseInfiniteQueryOptions,
+  type UseInfiniteQueryResult,
+  type UseMutationOptions,
+  type UseMutationResult,
+  type UseQueryOptions,
+  type UseQueryResult,
+  useInfiniteQuery as tanstack_useInfiniteQuery,
+  useQuery as tanstack_useQuery,
+  useMutation,
+} from '@tanstack/react-query'
+
+export type UseMutationParameters<
+  data = unknown,
+  error = Error,
+  variables = void,
+  context = unknown,
+> = Compute<
+  Omit<
+    UseMutationOptions<data, error, Compute<variables>, context>,
+    'mutationFn' | 'mutationKey' | 'throwOnError'
+  >
+>
+
+export type UseMutationReturnType<
+  data = unknown,
+  error = Error,
+  variables = void,
+  context = unknown,
+> = Compute<
+  UnionStrictOmit<
+    UseMutationResult<data, error, variables, context>,
+    'mutate' | 'mutateAsync'
+  >
+>
+
+export { useMutation }
+
+////////////////////////////////////////////////////////////////////////////////
+
+export type UseQueryParameters<
+  queryFnData = unknown,
+  error = DefaultError,
+  data = queryFnData,
+  queryKey extends QueryKey = QueryKey,
+> = Compute<
+  ExactPartial<
+    Omit<UseQueryOptions<queryFnData, error, data, queryKey>, 'initialData'>
+  > & {
+    // Fix `initialData` type
+    initialData?:
+      | UseQueryOptions<queryFnData, error, data, queryKey>['initialData']
+      | undefined
+  }
+>
+
+export type UseQueryReturnType<data = unknown, error = DefaultError> = Compute<
+  UseQueryResult<data, error> & {
+    queryKey: QueryKey
+  }
+>
+
+// Adding some basic customization.
+// Ideally we don't have this function, but `import('@tanstack/react-query').useQuery` currently has some quirks where it is super hard to
+// pass down the inferred `initialData` type because of it's discriminated overload in the on `useQuery`.
+export function useQuery<queryFnData, error, data, queryKey extends QueryKey>(
```

Comment:
As I see we don't use it anywhere, so maybe we just remove all these helpers for now?